changelog/unreleased/kong/fix_malloc_assertion_error.yml

```diff
@@ -0,0 +1,3 @@
+message: "fix malloc assertion error by replacing thread tcp server in reports_spec.lua."
+type: bugfix
+scope: Core
```

Comment:
@ADD-SP Is any modifcation in `spec` about test cases uncessary to write one changelog entry?

I noticed that in my previous PR, you comment that we dont need this in test cases.

Reply:
Only the user-facing changes need the changlog.